apollo-portal/src/main/java/com/ctrip/framework/apollo/portal/util/NamespaceBOUtils.java

```diff
@@ -0,0 +1,42 @@
+package com.ctrip.framework.apollo.portal.util;
+
+import com.ctrip.framework.apollo.core.ConfigConsts;
+import com.ctrip.framework.apollo.portal.controller.ConfigsExportController;
+import com.ctrip.framework.apollo.portal.entity.bo.NamespaceBO;
+import com.google.common.base.Joiner;
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * @author wxq
+ */
+public class NamespaceBOUtils {
+
+  /**
+   * copy from old {@link ConfigsExportController}.
+   * convert {@link NamespaceBO} to a file content.
+   * @return content of config file
+   */
+  public static String convert2configFileContent(NamespaceBO namespaceBO) {
+    List<String> fileItems = namespaceBO.getItems().stream().map(itemBO -> {
+      String key = itemBO.getItem().getKey();
+      String value = itemBO.getItem().getValue();
+
+      // special namespace format
+      if (ConfigConsts.CONFIG_FILE_CONTENT_KEY.equals(key)) {
+        return value;
+      }
+
+      if ("".equals(key)) {
+        return Joiner.on("").join(itemBO.getItem().getKey(), itemBO.getItem().getValue());
+      }
+
+      return Joiner.on(" = ").join(itemBO.getItem().getKey(), itemBO.getItem().getValue());
```

Comment:
Cause some problem when `itemBO.getItem().getValue()` exist `\n`.

Example:
http://106.54.227.205/config.html?#/appid=testAPP001
namespace `TEST1.ffffffff`,
exist `\n` in value, because it is a json.
So generate the wrong `properties` format in key `hello`.
```properties
test2 = gray0204-2020
hello = [
{"name":"jc", "age"=20, "male":true, "address":{"street":"21 science park", "city":"sg","country":"sg"}},
{"name":"jx", "age"=22, "male":false, "address":{"street":"22 science park", "city":"sg","country":"sg"}}
]
birthday = test
aaaa = bbbb
test_2020 = hello1
server.port = 8080
dafeiji = 123
keysfsf = lmc168
po = 1 3
pop = 1
yyy = 123
enable cloud = true
```

Reply:
I think we should put them into a Properties object and then transform to properties file, e.g.

https://github.com/ctripcorp/apollo/blob/6b6eb1835b5cf3c01dca8b0f0aa1dcff847117be/apollo-core/src/main/java/com/ctrip/framework/apollo/core/utils/PropertiesUtil.java#L17-L23

Reply:
The code is update. The method use `com.ctrip.framework.apollo.core.utils.PropertiesUtil#toString` now.